aten/src/ATen/native/cuda/BatchLinearAlgebra.cpp

```diff
@@ -3172,84 +3172,6 @@ void lstsq_kernel(const Tensor& a, Tensor& b, Tensor& /*rank*/, Tensor& /*singul
 
 REGISTER_CUDA_DISPATCH(lstsq_stub, &lstsq_kernel);
 
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ legacy_lstsq ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-std::tuple<Tensor, Tensor> legacy_lstsq_cuda(const Tensor &B, const Tensor &A) {
```

Comment:
This seems a bit odd. Did we reimplement all the calling code for `magmaGels` within `linalg_lstsq_cuda`? I guess we did because of the `TORCH_WARN_ONCE` that's here, but that seems so odd.

`lstsq` is a mess, it could do with a reasonable clean-up.

Reply:
Porting of TH version of `lstsq` to ATen was done independently from implementing `torch.linalg.lstsq`.

Reply:
Fantastic to get rid of this code then (same applies to all the other PRs, but to this one in particular).